### PR TITLE
tests: logging: syst: Add testcases for mipi syst.

### PR DESCRIPTION
**Pull request by aasthagr**
Adding testcases for mipi syst logging format.

Signed-off-by: Aastha Grover <aastha.grover@intel.com>

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -179,7 +179,8 @@ manifest:
       path: modules/debug/mipi-sys-t
       groups:
         - debug
-      revision: d9da086b11cda494d85f4d8a9829f505c2d5e380
+      url: https://github.com/aasthagr/mipi-sys-t
+      revision: syst_fix_add_wint_type
```
Comment (nashif) [inline west.yml:183]: not the right way to do this, please refer to the docs for how this should be done.

https://docs.zephyrproject.org/latest/guides/modules.html#process-for-submitting-changes-to-existing-modules

**Review by nashif — CHANGES_REQUESTED** [review]

File: west.yml (hunk 1 above)
Comment (aasthagr) [reply west.yml]: Sure. Updated this as per the docs.

**Comment by laurenmurphyx64** [comment]
Hey, I know we were having trouble figuring out how to get the string constants on one line without getting newlines inserted, but I was taking a look at a PR (#43984) Tom just put up and had the idea to do this:

```
#define PAYLOAD "4465627567206D657373616765206578616D706C652C2025642C2025642C202564000"\
	"10000000200000003000000"
const char *payload = PAYLOAD;
...
void test_main(void)
{
	printk("PAYLOAD: %s", PAYLOAD);
...
```

Which printed this:

```
Booting from ROM..*** Booting Zephyr OS build zephyr-v3.0.0-1284-g41805a779e05  ***

PAYLOAD 4465627567206D657373616765206578616D706C652C2025642C2025642C20256400010000000200000003000000

Running test suite test_log_syst...
```

And when I ran the compliance check CI does (see `.github/workflows/compliance.yml:72`), I got no warnings:

```
Running checkpatch       tests in /home/herbert/zephyr-external/zephyr ...
Running Kconfig          tests in /home/herbert/zephyr-external/zephyr ...
Running Codeowners       tests in /home/herbert/zephyr-external/zephyr ...
Running Nits             tests in /home/herbert/zephyr-external/zephyr ...
Running Gitlint          tests in /home/herbert/zephyr-external/zephyr ...
Running pylint           tests in /home/herbert/zephyr-external/zephyr ...
Running Identity         tests in /home/herbert/zephyr-external/zephyr ...

Complete results in compliance.xml
```

Examining the string in the debugger yields one with no breaks or extra characters (copied from Visual Studio Code debugger):

```
payload: 0x1054ac "4465627567206D657373616765206578616D706C652C2025642C2025642C20256400010000000200000003000000"
```

Hopefully this should make it so you don't have to build the string yourself.

**Comment by aasthagr** [comment]
> Hey, I know we were having trouble figuring out how to get the string constants in under the line length limit without getting newlines inserted, but I was taking a look at a PR (#43984) Tom just put up and had the idea to do this:
> 
> ```
> #define PAYLOAD "4465627567206D657373616765206578616D706C652C2025642C2025642C202564000"\
> 	"10000000200000003000000"
> const char *payload = PAYLOAD;
> ...
> void test_main(void)
> {
> 	printk("PAYLOAD: %s", payload);
> ...
> ```
> 
> Which printed this:
> 
> ```
> Booting from ROM..*** Booting Zephyr OS build zephyr-v3.0.0-1284-g41805a779e05  ***
> 
> PAYLOAD 4465627567206D657373616765206578616D706C652C2025642C2025642C20256400010000000200000003000000
> 
> Running test suite test_log_syst...
> ```
> 
> And when I ran the compliance check CI does (see `.github/workflows/compliance.yml:72`), I got no warnings:
> 
> ```
> Running checkpatch       tests in /home/herbert/zephyr-external/zephyr ...
> Running Kconfig          tests in /home/herbert/zephyr-external/zephyr ...
> Running Codeowners       tests in /home/herbert/zephyr-external/zephyr ...
> Running Nits             tests in /home/herbert/zephyr-external/zephyr ...
> Running Gitlint          tests in /home/herbert/zephyr-external/zephyr ...
> Running pylint           tests in /home/herbert/zephyr-external/zephyr ...
> Running Identity         tests in /home/herbert/zephyr-external/zephyr ...
> 
> Complete results in compliance.xml
> ```
> 
> Examining the string in the debugger yields one with no breaks or extra characters (copied from Visual Studio Code debugger):
> 
> ```
> payload: 0x1054ac "4465627567206D657373616765206578616D706C652C2025642C2025642C20256400010000000200000003000000"
> ```
> 
> Hopefully this should make it so you don't have to build the string yourself.

@laurenmurphyx64 I had already tried this but checkpatch did complaint atleast in my local environment. May be CI does not complain about it and checkpatch in local still does? I can try though. Thanks!!

File: subsys/logging/log_output_syst.c [hunk 2]
```diff
@@ -308,7 +308,7 @@ static void write_d32(struct mipi_syst_handle *systh, uint32_t v)
 }
 
 #if defined(MIPI_SYST_PCFG_ENABLE_64BIT_IO)
-static void write_d64(struct mipi_syst_handle *systh, uint64_t v)
```
Comment (nashif) [inline subsys/logging/log_output_syst.c:311]: this should be in its own commit explaining why you are changing this.

File: subsys/testsuite/Kconfig [hunk 3]
```diff
@@ -160,4 +160,17 @@ config TEST_BUSY_SIM
 	  It simulates cpu load by using counter device to generate interrupts
 	  with random intervals and random busy looping in the interrupt.
 
+config LOG_BACKEND_MOCK
```
Comment (nashif) [inline subsys/testsuite/Kconfig:163]: why are you defining this to be part of the testsuite and without any associated code? How can this be used? Maybe put this as part of the test itself if you need that or have it in its own commit and  give some details how this can be used.

File: tests/subsys/logging/log_syst/src/main.c [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+#endif
+	const char *payload_body =
+	"4465627567206D657373616765206578616D706C652C2025642C2025642C202564000";
+	const char *payload_params =
+	"10000000200000003000000";
+	char *payload = malloc(strlen(payload_body)+strlen(payload_params)+1);
+
+	strcpy(payload, "");
+	strcat(payload, payload_body);
+	strcat(payload, payload_params);
```
Comment (nashif) [inline tests/subsys/logging/log_syst/src/main.c:104]: I am not sure exactly why are you doing here and why you need all this instead of defining the string directly...

File: tests/subsys/logging/log_syst/src/main.c [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+#ifdef CONFIG_64BIT
+	sub_type = "0C";
+#else
+	sub_type = "0B";
+#endif
+	const char *payload_body =
+	"4465627567206D657373616765206578616D706C652C2025642C2025642C202564000";
+	const char *payload_params =
+	"10000000200000003000000";
+	char *payload = malloc(strlen(payload_body)+strlen(payload_params)+1);
```
Comment (nashif) [inline tests/subsys/logging/log_syst/src/main.c:100]: why is this needed?

File: tests/subsys/logging/log_syst/testcase.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
+tests:
+  logging.log_syst.mipi_syst:
+    tags: log_syst logging
+    arch_exclude: mips nios2 posix sparc
```
Comment (nashif) [inline tests/subsys/logging/log_syst/testcase.yaml:4]: explain why those are excluded

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/subsys/logging/log_syst/src/main.c [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+{
+
+#ifdef CONFIG_LOG_MIPI_SYST_ENABLE
+	uint32_t test_log_type_syst = LOG_OUTPUT_SYST;
+
+	log_format_func_t test_log_output_func = log_format_func_t_get(test_log_type_syst);
+
+	zassert_equal_ptr(test_log_output_func, log_output_msg2_syst_process,
+					"Correct Function pointer for SYST log\n"
+					"log format was not selected %x vs %x",
```
Comment (edersondisouza) [inline tests/subsys/logging/log_syst/src/main.c:46]: Why not `%p` to show the pointer value?

File: tests/subsys/logging/log_syst/src/main.c (hunk 5 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/main.c]: Removed this. Was doing this earlier to address the checkpatch warnings treated as error for long string. Have now used the macro instead.

File: tests/subsys/logging/log_syst/src/main.c (hunk 4 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/main.c]: Have removed this. Was using to address the checkpatch warning.

File: subsys/testsuite/Kconfig (hunk 3 above)
Comment (aasthagr) [reply subsys/testsuite/Kconfig]: Have moved this option to Kconfig under the testcase.

File: subsys/logging/log_output_syst.c (hunk 2 above)
Comment (aasthagr) [reply subsys/logging/log_output_syst.c]: Not that there is a filter in place for 64 bit architectures, this is not needed. Hence changed it back to uint64_t.

File: tests/subsys/logging/log_syst/src/main.c (hunk 7 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/main.c]: yeahh %p is probably better to use here. Changed to %p. Thanks!!

File: tests/subsys/logging/log_syst/prj.conf [hunk 8]
```diff
@@ -0,0 +1,21 @@
+CONFIG_LOG=y
+CONFIG_ASSERT=n
+CONFIG_ZTEST=y
+CONFIG_LOG_BACKEND_MOCK=y
+CONFIG_LOG_BACKEND_UART=n
+CONFIG_SOC_LOG_LEVEL_OFF=y
+CONFIG_ARCH_LOG_LEVEL_OFF=y
+CONFIG_KERNEL_LOG_LEVEL_OFF=y
+CONFIG_LOG_RUNTIME_FILTERING=n
+CONFIG_MAIN_THREAD_PRIORITY=5
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/prj.conf:10]: Do you really need to set the main thread priority ?

File: tests/subsys/logging/log_syst/prj.conf [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+CONFIG_LOG_FUNC_NAME_PREFIX_DBG=n
+
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/prj.conf:12]: Same thing, do you need to turn off function name prefix for this test ?

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+/**
+ * @brief This function takes in the characters and copies
+ * them in the temporary buffer.
+ *
+ * @return No. of characters copied to the temporary buffer.
+ */
+static int char_out(uint8_t *data, size_t length, void *ctx)
+{
+	ARG_UNUSED(ctx);
+	size_t output_length = length;
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/src/mock_backend.c:29]: you don't need this variable. You can just return length.

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+
+	while (length > 0) {
+		test_output_buf[i] = *data;
+		i++;
+		data++;
+		length--;
+
+		if (i == sizeof(test_output_buf)-1) {
+			i = 0;
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/src/mock_backend.c:38]: Is it the behavior you want ? Overwrite the buffer data is bigger than the space you have.

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+#include <logging/log_core.h>
+#include <logging/log_backend_std.h>
+#include <stdlib.h>
+
+static uint32_t log_format_current = CONFIG_LOG_BACKEND_MOCK_OUTPUT_DEFAULT;
+union log_msg2_generic *test_msg;
+
+static uint8_t mock_output_buf[1];
+uint8_t test_output_buf[256];
+int i;
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/src/mock_backend.c:18]: That is a bad name for a variable with this scope. Please, rename it.

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+void validate_msg(const char *type, const char *optional_flags,
+					const char *module_id, const char *sub_type,
+					const char *payload)
+{
+	const char *raw_data_str = "SYS-T RAW DATA: ";
+	const char *output_str = test_output_buf;
+	const char *syst_format_headers[4] = {type, optional_flags, module_id, sub_type};
+	const char *syst_headers_name[4] = {"type", "optional_flags", "module_id", "sub_type"};
+
+    /* Validate "SYS-T RAW DATA: " prefix in the output_str */
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/src/mock_backend.c:75]: wrong identation.

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+	size_t skip_bytes[5] = {0, 18, 4, 8, 16};
+
+	for (int i = 1; i < ARRAY_SIZE(skip_bytes); i++) {
+		if (isKthBitSet(strtol(optional_flags, 0, 16), i)) {
+			output_str += skip_bytes[i];
+		}
+	}
+
+	zassert_mem_equal(output_str, payload, strlen(payload),
+								  "Incorrect Comparison of payload");
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/src/mock_backend.c:108]: ditto

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+	 */
+	size_t skip_bytes[5] = {0, 18, 4, 8, 16};
+
+	for (int i = 1; i < ARRAY_SIZE(skip_bytes); i++) {
+		if (isKthBitSet(strtol(optional_flags, 0, 16), i)) {
+			output_str += skip_bytes[i];
+		}
+	}
+
+	zassert_mem_equal(output_str, payload, strlen(payload),
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/src/mock_backend.c:107]: does the payload always has a null terminator ? If not, you can't use `strlen()`

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+
+static void mock_init(struct log_backend const *const backend)
+{
+
+}
+
+static void panic(struct log_backend const *const backend)
+{
+
+}
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/src/mock_backend.c:127]: These two functions are not doing anything. Can't you just set `.init` and `.panic` to NULL ?

File: tests/subsys/logging/log_syst/src/mock_backend.h [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern const struct log_backend_api mock_log_backend_api;
+
+struct mock_log_backend_msg {
+	/* Not used. */
+	uint8_t data[32];
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/src/mock_backend.h:20]: What does this comment means ? If that is not used, what is used in `mock_log_backend` ? just `bool panic;` ?

File: tests/subsys/logging/log_syst/src/mock_backend.h [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+ * @param type Field in message header to indicate the type of SyS-T message.
+ * @param optional_flags Optional Flags in message header.
+ * @param module_id ModuleID in a Message Header distinguish between multiple
+ *					instances of the same Origin.
+ * @param sub_type Enumerated types for representing the sub-type of a Message.
+ * @param payload The content of the Message.
+ */
+void validate_msg(const char *type, const char *optional_flags,
+					const char *module_id, const char *sub_type,
+					const char *payload);
```
Comment (ceolin) [inline tests/subsys/logging/log_syst/src/mock_backend.h:40]: fix identation here as well.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 10 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c:29]: length is getting altered in char_out routine. output_length is to return the number of characters copied to the buffer.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 16 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c:127]: On Some platforms like native_posix, panic is called at the end of test to flush all outputs and if it finds NULL pointer to panic then that results in a segmentation fault. So it is necessary to define these functions even though they are empty.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 15 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c:107]: Since the strings are created/defined in main.c from where the validate_msg routine is being called , it will always implicitly have a null terminator.

File: tests/subsys/logging/log_syst/prj.conf (hunk 8 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/prj.conf]: Removed the CONFIG_MAIN_THREAD_PRIORITY for now.

File: tests/subsys/logging/log_syst/prj.conf (hunk 9 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/prj.conf:11]: The CONFIG_LOG_FUNC_NAME_PREFIX_DBG is needed as it results in modification of output str which is used for validation here.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 11 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c]: When the buffer is full, the buffer size needs to be increased. In our case this block would not have executed since we know the input strings passed from the testcase. But good point, replaced this logic with zassert statement which would ask to increase the size of buffer when it is full instead of overriding it.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 12 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c]: Renamed. Thanks!!

File: tests/subsys/logging/log_syst/src/mock_backend.h (hunk 18 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.h]: Fixed.

File: tests/subsys/logging/log_syst/src/mock_backend.h (hunk 17 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.h]: Removed the mock_log_backend_msg and its reference from mock_log_backend struct for now as mock backend is not holding any messages for now. Can add back later if its use is identified.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 14 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c]: Fixed

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 13 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c]: Fixed.

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+	test_msg = msg;
+
+	log_format_func_t log_output_func = log_format_func_t_get(log_format_current);
+
+	log_output_func(&log_output_mock, &msg->log, flags);
+}
+
+bool isKthBitSet(int n, int k)
+{
+	if (n & (1 << (k - 1)))
```
Comment (edersondisouza) [inline tests/subsys/logging/log_syst/src/mock_backend.c:59]: Nitpick: this could simply be `return (n & (1 << (k - 1)))`

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+		return true;
+	else
+		return false;
+}
+
+void validate_msg(const char *type, const char *optional_flags,
+					const char *module_id, const char *sub_type,
+					const char *payload)
```
Comment (edersondisouza) [inline tests/subsys/logging/log_syst/src/mock_backend.c:67]: I think that on function declaration, the goal is to indent the next line at the same column just after the opening parenthesis. Something like:
```
void validate_msg(const char *type, const char *optional_flags,
		  const char *module_id, const char *sub_type,
		  const char *payload)
```
Use tabs to get closer, and spaces to finish indenting.

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 21]
```diff
@@ -0,0 +1,135 @@
...
+				  "Incorrect Comparison of %s", syst_headers_name[i]);
+
+		output_str = output_str+2;
+	}
+
+	/* After the headers the output_str will contain the content of optional flags.
+	 * Optional flags are contained in 1st byte of data in output_str. There are 4 bits
+	 * reserved for Optional flags. 1st bit = Location, 2nd Bit = payload length
+	 * 3rd bit = Sys-t Checksum, 4th bit = Sys-t Timestamp. Support for validating the
+	 * content based on optional flags is not supported, hence the corrosponding
```
Comment (edersondisouza) [inline tests/subsys/logging/log_syst/src/mock_backend.c:93]: s/corrosponding/corresponding/

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 22]
```diff
@@ -0,0 +1,135 @@
...
+	 * characters are skipped. skip_bytes array contains the number of hex characters
+	 * to skip in the output_str.
+	 */
+	size_t skip_bytes[5] = {0, 18, 4, 8, 16};
+
+	for (int i = 1; i < ARRAY_SIZE(skip_bytes); i++) {
```
Comment (edersondisouza) [inline tests/subsys/logging/log_syst/src/mock_backend.c:99]: Why start on one? What about making `skip_bytes` have 4 items only?

File: scripts/ci/check_compliance.py [hunk 23]
```diff
@@ -561,6 +561,8 @@ def get_defined_syms(kconf):
     "REG2",
```
Comment (edersondisouza) [inline scripts/ci/check_compliance.py:561]: On commit message: s/complaince/compliance/g

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 22 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c:99]: Coz optional flags are contained in 1st byte of data. So when we do  _if (isKthBitSet(strtol(optional_flags, 0, 16), i))_ , i needs to start from 1. Either the logic can be tweaked to do i+1 here if skip_bytes contained 4 items or the way it is done works fine.

File: tests/subsys/logging/log_syst/CMakeLists.txt [hunk 24]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
```
[the remark this answers is not in the document]
Comment (aasthagr) [reply tests/subsys/logging/log_syst/CMakeLists.txt:1]: Fixed

File: tests/subsys/logging/log_syst/Kconfig [hunk 25]
```diff
@@ -0,0 +1,18 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2022 Intel Corporation
+
+config LOG_BACKEND_MOCK
+	bool "MOCK backend"
+	help
+	  When enabled backend is using MOCK backend to output logs.
```
[the remark this answers is not in the document]
Comment (aasthagr) [reply tests/subsys/logging/log_syst/Kconfig]: Modified this. Please re-review.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 19 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c]: yeahh, totally makes sense :) Changed.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 20 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c]: Done.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 21 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c]: Fixed

File: scripts/ci/check_compliance.py (hunk 23 above)
Comment (aasthagr) [reply scripts/ci/check_compliance.py:561]: Fixed

File: tests/subsys/logging/log_syst/src/main.c [hunk 26]
```diff
@@ -0,0 +1,151 @@
...
+
+	const char *type = "72";
+	const char *optional_flags = "0A";
+	const char *module_id = "00";
+	const char *sub_type;
+#ifdef CONFIG_64BIT
+	sub_type = "0C";
+#else
+	sub_type = "0B";
+#endif
```
Comment (peter-mitsis) [inline tests/subsys/logging/log_syst/src/main.c:77]: I note that there are a few places where this logic is duplicated. It may be worthwhile to do something like ...

#ifdef CONFIG_64BIT
    #define SUB_TYPE  "0C"
#else
    #define SUB_TYPE "0B"
#endif

And then in the test code ...
    const char *sub_type = SUB_TYPE;

This would have the advantage of having the conditional logic in one place only.

File: tests/subsys/logging/log_syst/src/mock_backend.c [hunk 27]
```diff
@@ -0,0 +1,132 @@
...
+	test_msg = msg;
+
+	log_format_func_t log_output_func = log_format_func_t_get(log_format_current);
+
+	log_output_func(&log_output_mock, &msg->log, flags);
+}
+
+bool isKthBitSet(int n, int k)
+{
+	return (n & (1 << (k - 1)));
```
Comment (peter-mitsis) [inline tests/subsys/logging/log_syst/src/mock_backend.c:59]: Very minor quibble. If returning a bool, this should be using something like ...

    return ((n & (1 << (k - 1))) != 0);

... otherwise it is returning an integer.

File: tests/subsys/logging/log_syst/src/main.c [hunk 28]
```diff
@@ -0,0 +1,151 @@
...
+	const char *payload =
+	"4465627567206D657373616765206578616D706C652C2025640001000000";
```
Comment (peter-mitsis) [inline tests/subsys/logging/log_syst/src/main.c:79]: This comment is a maintenance suggestion. Feel free to ignore.

I note that the payload in each of the tests is comprised of two hex-string parts--format string and arguments. As the hex-strings are somewhat harder to interpret, it may be worthwhile to have an intermediary validation routine that converts a string to a hex-string. Not only should this make it easier to follow what each payload test is verifying, it also provides some flexibility should the format strings change in the future.

File: tests/subsys/logging/log_syst/src/mock_backend.c (hunk 27 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/mock_backend.c]: Good catch. Thanks. Addressed!!

File: tests/subsys/logging/log_syst/src/main.c (hunk 26 above)
Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/main.c]: Makes sense. Have moved additional declarations which were duplicated outside of test functions.

**Review by peter-mitsis — DISMISSED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]

**Review by peter-mitsis — DISMISSED** [review]

File: tests/subsys/logging/log_syst/src/main.c [hunk 29]
```diff
@@ -0,0 +1,138 @@
...
+#include <logging/log_backend_std.h>
+#include <logging/log_ctrl.h>
+#include <logging/log_output.h>
+#include <tc_util.h>
+#include <stdbool.h>
+#include <ztest.h>
+#include <stdlib.h>
+
+#define TEST_MSG_0 "0 args"
+#define PAYLOAD_MULTIPLE_ARGS "4465627567206D657373616765206578616D706C652C2025642C2025642C20256"\
```
Comment (nordic-krch) [inline tests/subsys/logging/log_syst/src/main.c:26]: any chance of having a comment that would somehow explain this stream?
If it's STP stream maybe something like
`0x4 (D8) 0x46 0x5 (D16) ...` (not sure if i got it correct).

Comment (aasthagr) [reply tests/subsys/logging/log_syst/src/main.c:29]: @nordic-krch I have added the comment for PAYLOAD_MULTIPLE_ARGS explaining its purpose.

**Review by nordic-krch — APPROVED** [review]

**Review by peter-mitsis — APPROVED** [review]

**Review by laurenmurphyx64 — APPROVED** [review]

**Review by nashif — APPROVED** [review]
